src/app/components/quiz.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { useState } from "react";
+import clsx from "clsx";
+
+export interface QuizParams {
+  className?: string;
+  question: string;
+  hint?: string;
+  answers: string[];
+  correctAnswer: number;
+  hintAnswers?: string[];
+  onSelect?: (selection: number, isDone: boolean) => void;
+  // If enabled, the user only has a single try. After selection, it will show the solution to the user.
+  showCorrectAnswer?: boolean;
+}
+
+/**
+ * A quiz component.
+ * @param quiz The quiz parameters
+ */
```

Comment:
you don't need to document the component. the name of the component should already tell you what the document is about!